src/promptflow/promptflow/executor/_service/utils/process_utils.py

```diff
@@ -0,0 +1,116 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import asyncio
+import contextlib
+import json
+import multiprocessing
+import os
+from datetime import timedelta
+from multiprocessing import Queue
+from typing import Callable
+
+from promptflow._core._errors import UnexpectedError
+from promptflow._core.operation_context import OperationContext
+from promptflow._utils.exception_utils import ExceptionPresenter, JsonSerializedPromptflowException
+from promptflow._utils.logger_utils import service_logger
+from promptflow.exceptions import ErrorTarget
+from promptflow.executor._service._errors import ExecutionTimeoutError
+from promptflow.executor._service.utils.service_utils import get_log_context
+
+ASYNC_REQUEST_TIMEOUT = timedelta(days=1).total_seconds()
+SYNC_REQUEST_TIMEOUT = 10  # seconds
+WAIT_SUBPROCESS_EXCEPTION_TIMEOUT = 10  # seconds
+
+
+async def invoke_function_in_process(
+    request, context_dict: dict, target_function: Callable, is_async_call: bool = True
+):
+    with multiprocessing.Manager() as manager:
+        return_dict = manager.dict()
+        exception_queue = manager.Queue()
```

Comment:
Do we have bandwidth to improve the `manager.Queue()` also?
It's obviously we only retrieve one item from the queue.
And we just try to get item from it when sub process is finished, the timeout seems also not necessary.

Reply:
You mean we can pass a dict not a queue?

Reply:
Yes, dict looks better than queue, not sure if there are other options.

Reply:
Updated.